main.nf

```diff
@@ -0,0 +1,684 @@
+    """
+    fusioncatcher \\
+        -d ${data_dir} \\
+        -i ${option} \\
+        --threads ${task.cpus} \\
+        -o . \\
+        --skip-blat \\
+        ${params.fc_extra_options}
+    """
+}
+
+/*
+ * Ericscript
+ */
+process ericscript {
+    tag "$name"
+    publishDir "${params.outdir}/tools/Ericscript", mode: 'copy'
+
+    when:
+    params.ericscript && (!params.singleEnd || params.test)
+
+    input:
+    set val(name), file(reads) from read_files_ericscript
+    file reference from ericscript_ref.collect()
+
+    output:
+    file './tmp/fusions.results.total.tsv' into ericscript_fusions
+    file '*' into ericscript_output
+
+    script:
+    """
+    ericscript.pl \\
+        -db ${reference} \\
+        -name fusions \\
+        -p ${task.cpus} \\
+        -o ./tmp \\
+        ${reads[0]} \\
+        ${reads[1]}
+    """
+}
+
+/*
+ * Pizzly
+ */
+process pizzly {
+    tag "$name"
+    publishDir "${params.outdir}/tools/Pizzly", mode: 'copy'
+
+    when:
+    params.pizzly && (!params.singleEnd || params.test)
+
+    input:
+    set val(name), file(reads) from read_files_pizzly
+    file fasta from pizzly_fasta.collect()
+    file gtf from pizzly_gtf.collect()
+    
+    output:
+    file '*.unfiltered.json' into pizzly_fusions
+    file '*.{json,txt,tsv,fasta}' into pizzly_output
+
+    script:
+    """
+    kallisto index -i index.idx -k ${params.pizzly_k} ${fasta}
+    kallisto quant -t ${task.cpus} -i index.idx --fusion -o output ${reads[0]} ${reads[1]}
+    pizzly -k ${params.pizzly_k} \\
+        --gtf ${gtf} \\
+        --cache index.cache.txt \\
+        --align-score 2 \\
+        --insert-size 400 \\
+        --fasta ${fasta} \\
+        --output pizzly_fusions output/fusion.txt
+    """
+}
+
+/*
+ * Squid
+ */
+process squid {
+    tag "$name"
+    publishDir "${params.outdir}/tools/Squid", mode: 'copy'
+
+    when:
+    params.squid && (!params.singleEnd || params.test)
+
+    input:
+    set val(name), file(reads) from read_files_squid
+    file index from star_index.collect()
+    file gtf
+    
+    output:
+    file '*_annotated.txt' into squid_fusions
+    file '*' into squid_output
+
+    script:
+    def avail_mem = task.memory ? "--limitBAMsortRAM ${task.memory.toBytes() - 100000000}" : ''
+    """
+    STAR \\
+        --genomeDir ${index} \\
+        --sjdbGTFfile ${gtf} \\
+        --runThreadN ${task.cpus} \\
+        --readFilesIn ${reads[0]} ${reads[1]} \\
+        --twopassMode Basic \\
+        --chimOutType SeparateSAMold --chimSegmentMin 20 --chimJunctionOverhangMin 12 --alignSJDBoverhangMin 10 --outReadsUnmapped Fastx --outSAMstrandField intronMotif \\
+        --outSAMtype BAM Unsorted ${avail_mem} \\
+        --readFilesCommand zcat
+    samtools sort Aligned.out.bam > Aligned.out.sorted.bam
+    samtools view -Shb Chimeric.out.sam > Chimeric.out.bam
+    squid -b Aligned.out.sorted.bam -c Chimeric.out.bam -o fusions
+    AnnotateSQUIDOutput.py ${gtf} fusions_sv.txt fusions_annotated.txt
```

Comment:
Can we use `$name` in all of these file paths to make them specific to the sample?

Reply:
Done